### DISCUSSION TITLE
Experiences latency with zustand

### DISCUSSION THREAD

**Discussion by alexander-densley** [pmndrs/zustand#2465] [Q&A]
i am creating a form and i wanted to clean up code by taking logic for each part of the form into it's own componenent as the form component was getting to be 400-500 lines, and so to fix this i made each element its own component. I then used zustand to store the data for each of these components. however, when i now type in these elements, the typing is slightly delayed. when i switch back tousing usestate in the big form, it's instant. i can share a gist with what i was doing if anyone wants to take alook. I imagine its my implementation and not the library itself, thanks

**Comment by dbritto-dev** [comment]
@alexander-densley would you mind sharing you gist?

**Reply by alexander-densley** [reply]
https://gist.github.com/alexander-densley/5e125304ff011b3aea72fed6014cba4e

**Reply by alexander-densley** [reply]
the event-title component is where the typing is occuring, and the component is included in the modal.
